controller/app/cmdline/src/cmd_line.cpp

```diff
@@ -1382,30 +1382,34 @@ int cmd_line::do_select(uint32_t new_end_station, uint16_t new_entity, uint16_t
         uint16_t current_config = end_station->get_current_config_index();
         avdecc_lib::entity_descriptor_response * entity_desc_resp = end_station->get_entity_desc_by_index(current_entity)->get_entity_response();
 
-        uint8_t * end_station_name = entity_desc_resp->entity_name();
-
         if ((current_end_station == new_end_station) && (current_entity == new_entity) && (current_config == new_config))
         {
-            atomic_cout << "Same setting" << std::endl;
-            atomic_cout << "\tEnd Station Index: " << std::dec << current_end_station << " (" << end_station_name << ")" << std::endl;
-            atomic_cout << "\tEntity Index: " << std::dec << current_entity << std::endl;
+            atomic_cout << "Same selection" << std::endl;
+            atomic_cout << "\tEnd Station Index:   " << std::dec << current_end_station << std::endl;
+            atomic_cout << "\tEnd Station:         " << entity_desc_resp->entity_name()
+                        << " (0x" << std::setw(16) << std::hex << std::setfill('0') << end_station->entity_id()
+                        << ")" << std::endl;
+            atomic_cout << "\tEntity Index:        " << std::dec << current_entity << std::endl;
             atomic_cout << "\tConfiguration Index: " << std::dec << current_config << std::endl;
         }
         else
         {
             current_end_station = new_end_station;
             end_station->set_current_entity_index(new_entity);
             end_station->set_current_config_index(new_config);
-            atomic_cout << "New setting" << std::endl;
-            atomic_cout << "\tEnd Station Index: " << std::dec << current_end_station << " (" << end_station_name << ")" << std::endl;
-            atomic_cout << "\tEntity Index: " << std::dec << end_station->get_current_entity_index() << std::endl;
-            atomic_cout << "\tConfiguration Index: " << std::dec << end_station->get_current_config_index() << std::endl;
+            atomic_cout << "New selection" << std::endl;
+            atomic_cout << "\tEnd Station Index:   " << std::dec << current_end_station << std::endl;
+            atomic_cout << "\tEnd Station:         " << entity_desc_resp->entity_name()
+                        << " (0x" << std::setw(16) << std::hex << std::setfill('0') << end_station->entity_id()
+                        << ")" << std::endl;
+            atomic_cout << "\tEntity Index:        " << std::dec << current_entity << std::endl;
```

Comment:
```current_entity``` is a local assigned to the previous entity index - we want to print the new one.